regparser/web/jobs/models.py

```diff
@@ -25,3 +33,28 @@ class ParsingJob(models.Model):
         ("complete_with_errors", "complete_with_errors")
     ), default="received")
     url = models.URLField(blank=True, max_length=2000)
+
+    def save(self, *args, **kwargs):
+        super(ParsingJob, self).save(*args, **kwargs)
+
+
+class PipelineJob(ParsingJob):
+
+    cfr_title = models.IntegerField()
+    cfr_part = models.IntegerField()
+    only_latest = models.BooleanField(default=False)
+
+
+class ProposalPipelineJob(ParsingJob):
+
+    file_hexhash = models.CharField(max_length=32)
+    only_latest = models.BooleanField(default=True)
+
+
+class RegulationFile(models.Model):
+
+    contents = models.BinaryField()
+    file = models.FileField(null=True)
+    filename = models.CharField(default=None, max_length=512, null=True)
+    hexhash = models.CharField(default=None, max_length=32, null=True)
```

Comment:
Quite possibly. I was looking at the hashing approach as speculative, but if we keep it, then this should probably the the primary key.